src/main.rs

```diff
@@ -95,6 +102,58 @@ fn publish(request: Json<PublishRequest>, auth: TokenAuth) -> ApiResult<EmptyRes
     Ok(Json(EmptyResponse))
 }
 
+#[post(
+    "/upload_project?<forc_version>",
+    format = "application/gzip",
+    data = "<tarball>"
+)]
+async fn upload_project(
+    db: &State<Database>,
+    pinata_client: &State<PinataClientImpl>,
+    forc_version: &str,
+    mut tarball: TempFile<'_>,
+) -> ApiResult<UploadResponse> {
+    // Install the forc version if it's not already installed.
+    let forc_path_str = format!("forc-{forc_version}");
+    let forc_path = PathBuf::from(&forc_path_str);
+    fs::create_dir_all(forc_path.clone()).unwrap();
```

Comment:
could also use the [`tempfile`]() crate which handles creating temporary files and directories.
Not a fan of introducing external deps if not needed; but the good thing about the crate - and [`tempdir`](https://docs.rs/tempfile/latest/tempfile/fn.tempdir.html) function is that if an error occurs anywhere in the function before the directory is removed, the directory would still be removed.
This is because the dir removal is based on the `Drop` trait impl - hence dir is removed as soon as the var goes out of scope.
^ this would ensure clean future runs of the function (assuming a previous run failed)

Reply:
done in 5ed250f6d621d04d7272fab5bd27a567bd86ecf3